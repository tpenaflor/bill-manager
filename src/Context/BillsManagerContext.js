import React from 'react'

const BillsManagerContext = React.createContext()

export default BillsManagerContext