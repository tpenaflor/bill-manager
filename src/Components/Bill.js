import React from 'react'
import SplitAssignment from './SplitAssignment'
import {useBillsManagerContext} from '../Context/BillsManagerContext'

import {totalBillAllocation} from '../Selector/billSelector'

const Bill = ({bill}) => {
    const {billDispatch} = useBillsManagerContext()

    const removeBill = (id) => {
        billDispatch({type:"DEL_BILL", id})
    }

    return (
        <div>
            <button onClick={()=>removeBill(bill.id)}>x</button>
            <div>
                {bill.desc} 
                {bill.splitOpt === 'unit' && ` ${totalBillAllocation(bill)}/${bill.allocation.unitCount}`}
            </div>
            <div>{bill.amount}</div>
            <SplitAssignment bill={bill}/>
        </div> 
    )
}


export default Bill