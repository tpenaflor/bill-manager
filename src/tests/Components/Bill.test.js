import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';

import Bill from '../../Components/Bill'
import * as BillsManagerContext from '../../Context/BillsManagerContext'
import {billFixture} from '../Fixtures/BillFixture'

let billDispatch, wrapper

beforeEach( ()=> {
    billDispatch = jest.fn()
    const contextValues = {
        billDispatch
    }

    jest.spyOn(BillsManagerContext, 'useBillsManagerContext')
        .mockImplementation(() => contextValues)

    wrapper = shallow(<Bill bill={billFixture[0]}/>)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})

afterEach(()=>{
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('delete bill', ()=> {
    wrapper.find('button').at(0).simulate('click')

    expect(billDispatch).toHaveBeenLastCalledWith({
        type:"DEL_BILL", 
        id : billFixture[0].id
      })

})