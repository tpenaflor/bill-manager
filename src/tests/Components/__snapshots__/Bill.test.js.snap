// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete bill 1`] = `
<div
  className="entry"
>
  <div>
    <button
      onClick={[Function]}
    >
      x
    </button>
    beer
     (0/10)
     PHP 1,000.00
  </div>
  <SplitAssignment
    bill={
      Object {
        "allocation": Object {
          "participant": Array [],
          "unitCount": "10",
        },
        "amount": "1000",
        "desc": "beer",
        "id": "4efa1ac1-2d6f-49a5-834b-d4d5f0e7c97c",
        "splitOpt": "unit",
      }
    }
  />
</div>
`;

exports[`delete bill 2`] = `
<div
  className="entry"
>
  <div>
    <button
      onClick={[Function]}
    >
      x
    </button>
    beer
     (0/10)
     PHP 1,000.00
  </div>
  <SplitAssignment
    bill={
      Object {
        "allocation": Object {
          "participant": Array [],
          "unitCount": "10",
        },
        "amount": "1000",
        "desc": "beer",
        "id": "4efa1ac1-2d6f-49a5-834b-d4d5f0e7c97c",
        "splitOpt": "unit",
      }
    }
  />
</div>
`;
